bigtable-hbase-1.x-parent/bigtable-hbase-1.x-mapreduce/src/main/java/com/google/cloud/bigtable/mapreduce/validation/RecordFormatter.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.bigtable.mapreduce.validation;
+
+import com.google.cloud.bigtable.mapreduce.validation.SyncTableValidationOnlyJob.SyncMapper;
+import com.google.gson.JsonObject;
+import org.apache.hadoop.hbase.Cell;
+import org.apache.hadoop.hbase.KeyValue;
+import org.apache.hadoop.hbase.util.Bytes;
+
+/** helper class for formatting validation records for output */
+public class RecordFormatter {
+
+  public static final String REASON_CODE_STR = "reason_code";
+
+  /**
+   * formatting for range mismatch
+   * @param reasonCode
+   * @param startRow
+   * @param stopRow
+   * @param sourceHash
+   * @param targetHash
+   * @return
+   */
+  public static String formatRangeMismatch(
+      SyncMapper.Counter reasonCode,
```

Comment:
why is reasonCode a type of "SyncMapper.Counter"? Should it be String, or do we need a more expressive name?